docs/HowtoTDE.md

```diff
@@ -0,0 +1,219 @@
+
+```bash
+vault kv put v2.nuodb.com/TDE/demo \
+  target=topSecret \
+  historical-20201204=superSecret \
+  historical-20201203=secret
+
+```
+The secret name can be different from the example above.
+In this case _v2_ illustrates that Vault KV store version 2 is used and _v2.nuodb.com/TDE_ path can group several Vault secrets for several NuoDB databases.
+
+Vault policy should be created and attached to Vault Kubernetes role to allow `read` capability for the secret created above by NuoDB service account. In the examples further in this section, `nuodb` Vault Kubernetes role is used.
+
+The `tde.json` configuration file can be injected using the following Helm values file snippet:
+
+```yaml
+...
+
+admin:
+  podAnnotations:
+    vault.hashicorp.com/agent-inject: "true"
+    vault.hashicorp.com/role: nuodb
+    vault.hashicorp.com/agent-inject-file-demo: tde.json
+    vault.hashicorp.com/secret-volume-path-demo: /etc/nuodb/tde/demo
+    vault.hashicorp.com/agent-inject-secret-demo: v2.nuodb.com/TDE/demo
+    vault.hashicorp.com/agent-inject-template-demo: |
+      {{- with secret "v2.nuodb.com/TDE/demo" -}}
+      {{- .Data.data | toJSON }}
+      {{- end -}}
+ 
+database:
+  podAnnotations:
+    vault.hashicorp.com/agent-inject: "true"
+    vault.hashicorp.com/role: nuodb
+    vault.hashicorp.com/agent-inject-file-demo: tde.json
+    vault.hashicorp.com/secret-volume-path-demo: /etc/nuodb/tde/demo
+    vault.hashicorp.com/agent-inject-secret-demo: v2.nuodb.com/TDE/demo
+    vault.hashicorp.com/agent-inject-template-demo: |
+      {{- with secret "v2.nuodb.com/TDE/demo" -}}
+      {{- .Data.data | toJSON }}
+      {{- end -}}
+
+...
+```
+
+Templates for Vault KV store version 1 and version 2 are different..
+If Vault KV store version 1 is used, `.Data.data` references in inject templates should be replaced with `.Data`.
+
+> **NOTE**: For information on how different annotations can be used, see [here](https://www.vaultproject.io/docs/platform/k8s/injector/annotations).
+
+If `tde.conf` needs to be created using several secrets or special keys from the secret needs to be filtered out, [scratch](https://github.com/hashicorp/consul-template#scratch) Consul template API functions can be used. 
+For example this template will filter out any other keys in a Vault secret:
+
+```yaml
+    vault.hashicorp.com/agent-inject-template-demo: |
+      {{- with secret "v2.nuodb.com/TDE/demo" -}}
+      {{- range $k, $v := .Data.data -}}
+      {{- if or (eq $k "target") ($k | regexMatch "historical.*") }}
+      {{- scratch.MapSet "passwords" $k $v }}
+      {{- end }}
+      {{- end }}
+      {{- scratch.Get "passwords" | toJSON }}
+      {{- end -}}
+```
+
+
+## Enable Transparent Data Encryption
+
+NuoDB Admin will configure supplied storage passwords and will propagate them to engines.
+To verify that the target password is configured correctly, the following command should succeed returning no errors.
```

Comment:
the end of the sentence is wonky. I will need a native speaker to help me out here. @jleslie85 ?

```
the following command should succeed by returning no errors
```
?